src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverterTests.Object.cs

```diff
@@ -311,6 +311,72 @@ public override void Write(Utf8JsonWriter writer, object value, JsonSerializerOp
             }
         }
 
+        private class PrimitiveConverter : JsonConverter<object>
+        {
+            public override bool CanConvert(Type typeToConvert)
+                => true;
+
+            public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType == JsonTokenType.True)
+                {
+                    return true;
+                }
+
+                if (reader.TokenType == JsonTokenType.False)
+                {
+                    return false;
+                }
+
+                if (reader.TokenType == JsonTokenType.Number)
+                {
+                    if (reader.TryGetInt32(out int i))
+                    {
+                        return i;
+                    }
+
+                    return reader.GetDouble();
+                }
+
+                if (reader.TokenType == JsonTokenType.String)
+                {
+                    if (reader.TryGetDateTime(out DateTime datetime))
+                    {
+                        return datetime;
+                    }
+
+                    return reader.GetString();
+                }
+
+                throw new JsonException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, object value, JsonSerializerOptions options)
+            {
+                throw new InvalidOperationException("Should not get here.");
+            }
+        }
+        private class ClassWithPrimitives
```

Comment:
Can you add and test another version of this class where the value-type properties are nullable?

I'd expect `InvalidOperationException` to (already) be thrown for these cases.

Adding this test variant will help us close https://github.com/dotnet/runtime/issues/36329 as well.

Reply:
Added a test with nullables an added `[ActiveIssue("https://github.com/dotnet/runtime/issues/36329")]`